src/compiler/sys.ts

```diff
@@ -480,7 +480,7 @@ namespace ts {
         const newTime = modifiedTime.getTime();
         if (oldTime !== newTime) {
             watchedFile.mtime = modifiedTime;
-            watchedFile.callback(watchedFile.fileName, getFileWatcherEventKind(oldTime, newTime));
+            watchedFile.callback(watchedFile.fileName, getFileWatcherEventKind(oldTime, newTime), modifiedTime);

```

Comment:
We are now passing modifiedTime whenever we have it so during `tsc --build --watch` mode we dont need to re-query it